Automattic/wp-calypso
# Framework: Upgrade ESLint no-redeclare rule to error

Closes #8399

This pull request seeks to upgrade our [ESLint `no-redeclare` rule](http://eslint.org/docs/rules/no-redeclare) from a warning to error so that it is captured by our CircleCI integration and pre-commit hooks. Because our configuration extends [eslint-config-wpcalypso](https://github.com/Automattic/eslint-config-wpcalypso), its omission will cause it to fallback to the default value ([source](https://github.com/Automattic/eslint-config-wpcalypso/blob/300e60b6f943a4b3d4d0c97844188583b4023886/index.js#L53)). There were a few existing errors in the codebase which were simple enough fixes (included).

__Implementation Notes:__

Pings for authors of original code to verify:

- @allendav `client/me/two-step/index.jsx` (4864-gh-calypso-pre-oss)
- @dbspringer `client/my-sites/ads/form-earnings.jsx` (10410-gh-calypso-pre-oss)
- @dbspringer `client/my-sites/ads/form-settings.jsx` (10411-gh-calypso-pre-oss)

__Testing Instructions:__

Verify that there are no regressions on screens making use of modules affected by these changes.

Ensure that linting passes:

```
npm run lint
```

cc @SpenUK
